### PR TITLE
Fix : ComponentLine copy fixed

### PR DESCRIPTION
**Pull request by AlexisDieu**
- ComponentLine was using the same buffer than the one copied : fixed
- ComponentLine and particle were adding themself to a list in the scene when being copied but it was done during the copy (ctrl+C) and that should be done during the paste (ctrl+V or D) esle the list was incremented without the GameObject existing

### CODE REVIEW

**Review by AuxPort66 — APPROVED** [review]
Good job you even catch a memory leak.

File: Source/DataModels/Components/ComponentLine.cpp [hunk 1]
```diff
@@ -32,13 +32,24 @@ ComponentLine::ComponentLine(const bool active, GameObject* owner) : Component(C
 		});
 }
 
+ComponentLine::ComponentLine(const ComponentLine& other, GameObject* owner) :
```
Comment (PaablooCH) [inline Source/DataModels/Components/ComponentLine.cpp:35]: why You create the constructor with a second parameter?

Comment (AlexisDieu) [reply Source/DataModels/Components/ComponentLine.cpp:35]: this si the copy constructor, the first one is for the copy and the second one is to follow the Component(ComponentType::LINE, other.IsEnabled(), owner, true) so the Component constructor. Maybe other.getOwner() was good but I just follow the step I did for the default constructor

Comment (PaablooCH) [reply Source/DataModels/Components/ComponentLine.cpp:35]: what I mean is those steps are now duplicated and You should use the copy constructor of the `component` class like in the `componentTransform` class for example.
this is what copy constr does in `compoent` class:
`Component::Component(const Component& component) : type(component.type),
	enabled(component.enabled),
	owner(nullptr),
	canBeRemoved(component.canBeRemoved)
{
}`
and before you leave the function `copyComponent` in `gameobject.cpp` (line 484) you set the correct owner, so you are setting it twice:
image

Comment (AlexisDieu) [reply Source/DataModels/Components/ComponentLine.cpp]: I change it, thanks for the feedback

**Pull request merged**